annotations/src/lib.rs

```diff
@@ -65,14 +65,53 @@ pub enum TagPropagation {
     UninterpretedCall,
 }
 
-/// Provide a way to create tag transfer masks. It is equivalent to bitwise-or of all its arguments.
+/// Provide a way to create tag propagation sets. It is equivalent to bitwise-or of all its arguments.
 #[macro_export]
 macro_rules! tag_propagation_set {
     ($($x:expr),*) => {
         0 $(| (1 << ($x as u8)))*
     };
 }
 
+/// A tag propagation set indicating a tag is propagated by all operations.
+pub const TAG_PROPAGATION_ALL: TagPropagationSet = tag_propagation_set!(
+    TagPropagation::Add,
+    TagPropagation::AddOverflows,
+    TagPropagation::And,
+    TagPropagation::BitAnd,
+    TagPropagation::BitNot,
+    TagPropagation::BitOr,
+    TagPropagation::BitXor,
+    TagPropagation::Cast,
+    TagPropagation::Div,
+    TagPropagation::Equals,
+    TagPropagation::GreaterOrEqual,
+    TagPropagation::GreaterThan,
+    TagPropagation::IntrinsicBinary,
+    TagPropagation::IntrinsicBitVectorUnary,
+    TagPropagation::IntrinsicFloatingPointUnary,
+    TagPropagation::LessOrEqual,
+    TagPropagation::LessThan,
+    TagPropagation::LogicalNot,
+    TagPropagation::Mul,
+    TagPropagation::MulOverflows,
+    TagPropagation::Ne,
+    TagPropagation::Neg,
+    TagPropagation::Or,
+    TagPropagation::Offset,
+    TagPropagation::Rem,
+    TagPropagation::Shl,
+    TagPropagation::ShlOverflows,
+    TagPropagation::Shr,
+    TagPropagation::ShrOverflows,
+    TagPropagation::Sub,
+    TagPropagation::SubOverflows,
+    TagPropagation::UninterpretedCall
+);
+
+/// A pre-defined tag for constant-time security. It is propagated by all operations.
+pub struct ConstantTimeTaint {}
```

Comment:
I am surprised that this type gets special treatment. In general, I am not really comfortable with only this type leading to constant time diagnostics. Let's discuss some more over VC.